pkg/server/node.go

```diff
@@ -651,17 +653,52 @@ func (n *Node) start(
 		return errors.Wrapf(err, "couldn't gossip descriptor for node %d", n.Descriptor.NodeID)
 	}
 
-	// Create stores from the engines that were already initialized.
-	for _, e := range state.initializedEngines {
-		s := kvserver.NewStore(ctx, n.storeCfg, e, &n.Descriptor)
-		if err := s.Start(workersCtx, n.stopper); err != nil {
-			return errors.Wrap(err, "failed to start store")
+	// Create stores from engines that are already initialized. This uses a channel to collect errors
+	// when starting stores in separate goroutines. The channel is a buffered channel with the same size as the number of
+	// stores so worker go routines will never wait. By default, stores will be started concurrently. To start stores
+	// sequentially set the environment variable COCKROACH_CONCURRENT_STORE_START=false
+	var sem *quotapool.IntPool
+	var startStoresAsync = envutil.EnvOrDefaultBool("COCKROACH_CONCURRENT_STORE_START", true)
```

Comment:
Let's move this up to a package global, below the "Cluster settings" block at the top of this file, along with the corresponding comment.